From e58243965bbc3a5f33166d650529926a10bcef30 Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Thu, 29 Jul 2021 14:59:06 +0530
Subject: [PATCH] Add DRM_FORMAT_NV21 flex layout update in lockYCbCr

Some of the camera apps are requesting for NV21 streams.
As lockYCbCr fails with not supported, preview is not seen
with those apps.

Fix the issue by adding DRM_FORMAT_NV21 flex layoyt update
in

Note: Inorder to support apps requiring NV21 support, framework
and camera vHAL changes are required along with this change.

Change-Id: I625764149f1bd23298a9f4e58f756bd3a8adaa01
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 cros_gralloc/gralloc1/cros_gralloc1_module.cc | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/cros_gralloc/gralloc1/cros_gralloc1_module.cc b/cros_gralloc/gralloc1/cros_gralloc1_module.cc
index 2be7992..a1a8d9a 100755
--- a/cros_gralloc/gralloc1/cros_gralloc1_module.cc
+++ b/cros_gralloc/gralloc1/cros_gralloc1_module.cc
@@ -842,6 +842,14 @@ int32_t CrosGralloc1::lockYCbCr(buffer_handle_t bufferHandle,
 		ycbcr->cstride = hnd->strides[1];
 		ycbcr->chroma_step = 4;
 		break;
+	case DRM_FORMAT_NV21:
+		ycbcr->y = addr[0];
+		ycbcr->cb = addr[1] + 1;
+		ycbcr->cr = addr[1];
+		ycbcr->ystride = hnd->strides[0];
+		ycbcr->cstride = hnd->strides[1];
+		ycbcr->chroma_step = 2;
+		break;
 	default:
 		return CROS_GRALLOC_ERROR_UNSUPPORTED;
 	}
-- 
2.17.1

